### ISSUE TITLE
`Scalene error: received signal SIGABRT`

### ISSUE THREAD

**Issue by thissasbnk** [plasma-umass/scalene#181]
Hello everyone, 

I got this message **Scalene error: received signal SIGABRT** 
what does it mean and what should i do ?

Thank you in advance for your answers

**Comment by sternj** [comment]
Can you please provide an example that we can run of this happening?

**Comment by sternj** [comment]
It looks like your computer is running out of memory! This doesn't appear to be a Scalene issue, but you might want to try running your project on a smaller input with Scalene to figure out where your memory issues are coming from.

**Comment by emeryberger** [comment]
Just to make sure: what platform are you running on, @thissasbnk ? Mac? Linux? Versions would be helpful, too (did you install Scalene via pip? What version?) Thanks!

**Comment by thissasbnk** [comment]
> Just to make sure: what platform are you running on, @thissasbnk ? Mac? Linux? Versions would be helpful, too (did you install Scalene via pip? What version?) Thanks!

hello, i'm using  Ubuntu 20.04 as a virtual box on Windows 10 with  scalene version 1.3.3 ( pip install --user --upgrade scalene)

**Comment by thissasbnk** [comment]
> You probably do need to increase the memory for your VM.
> That said, if execution runs long enough, you could have Scalene output results periodically, as in:
> % scalene --profile-interval 1 --outfile profile.html --html yourprogramhere.py
> This will update the file `profile.html` every second with the latest results. Hope this helps!

Hello, i don't think that it is related to memory, we have tried it on a 60Go machine, it still blocked at a certain moment
i've also tried this line % scalene --profile-interval 1 --outfile profile.html --html yourprogramhere.py, but i've got the same error message.
After this, we thought it was also related to parallelism but at the end we had the same error.
I'm desperate..

**Comment by emeryberger** [comment]
So quick question: did you adjust the memory allocated to the VM? The VM may not take advantage of the available physical memory if you don't specifically request it.

Failing that, if there's a way for you to provide the Python code & data so that we can attempt to reproduce the issue, that would be helpful.

**Comment by thissasbnk** [comment]
Hello, 
thank for your answer 
yes,i did unfortunately. The python code is available here : 
 https://gitlab.inria.fr/edebreuv/nutrimorph/-/blob/master/nutrimorph.py 

I can send you the data by email,
Thank you in advance

**Comment by emeryberger** [comment]
Sounds good: my e-mail is emery.berger@gmail.com. Thanks!

**Comment by emeryberger** [comment]
Just a note: there's not enough code here to be able to reproduce the problem.

**Comment by emeryberger** [comment]
Bump @thissasbnk

**Comment by emeryberger** [comment]
Tentatively closing this since there's not enough info to reproduce the problem. Hopefully the latest changes to Scalene also fix this issue. Please re-open if that's not the case and if you can provide us the missing code. Thanks!

**Comment by 59e5aaf4** [comment]
Hi, I suspect there's some kind of (virtual?) memory leak in scalene, it books memory at a decent pace (i'd say roughly more than 100MB/s ?) a while it reserves terabytes of RAM 🤣 

![image](https://user-images.githubusercontent.com/32863380/120439617-01026b80-c383-11eb-8f2a-1f4eb70e0179.png)

![image](https://user-images.githubusercontent.com/32863380/120439658-0cee2d80-c383-11eb-864a-1630bfa24d41.png)

![image](https://user-images.githubusercontent.com/32863380/120439756-2abb9280-c383-11eb-85c9-7eafa44a6645.png)

![image](https://user-images.githubusercontent.com/32863380/120439973-68b8b680-c383-11eb-9047-dd3fb0cf8ae6.png)

I understand that virtual memory might just be allocated and never touched, but surely at some point we're hitting a limit of the OS lol.

No thing in my `/var/log/*`, there's only the Scalene output: 

```
Scalene: internal error = 12 (src/include/samplefile.hpp:53)
Scalene error: received signal SIGABRT
```

I unfortunately can't share the code I'm using, but basically I'm instanciating lots of small objects, and doing intense loops on said objects which come from parsed XML.

It died when approaching the 8TB of RAM.

**Comment by emeryberger** [comment]
Which version of Scalene?

**Comment by 59e5aaf4** [comment]
> Which version of Scalene?

1.3.4.7 , installed from pypi

```
$ python3 -m pip show scalene
Name: scalene
Version: 1.3.4.7
Summary: Scalene: A high-resolution, low-overhead CPU, GPU, and memory profiler for Python
Home-page: https://github.com/emeryberger/scalene
Author: Emery Berger
Author-email: emery@cs.umass.edu
License: Apache License 2.0
Location: /usr/local/lib/python3.9/dist-packages
Requires: cloudpickle, numpy, nvidia-ml-py, rich
Required-by: 

$ python3 -m scalene --version
Scalene version 1.3.4.7

Python info: 
python version : 3.9.2 (default, Feb 28 2021, 17:03:44), [GCC 10.2.1 20210110]
platform       : Linux-5.8.0-1-amd64-x86_64-with-glibc2.31

Python package 3.9.2-3 on debian sid
```

**Comment by Akashdesarda** [comment]
I am also facing the same issue. I was constantly monitoring my memory usage & it was always in range of 3.5-3.8 gb out of 16 gb. So I don't think so it is a memory issue. I am though doing a lot of IO operation like first downloading and then writing almost 2200+ files. 

```
Name: scalene
Version: 1.3.4.7
Summary: Scalene: A high-resolution, low-overhead CPU, GPU, and memory profiler for Python
Home-page: https://github.com/emeryberger/scalene
Author: Emery Berger
Author-email: emery@cs.umass.edu
License: Apache License 2.0
Location: /home/akash.desarda@DuffandPhelps.com/miniconda3/lib/python3.9/site-packages
Requires: cloudpickle, nvidia-ml-py, rich, numpy
```

**Comment by emeryberger** [comment]
Just pushed a new version to pip (1.3.5). Please update and try again (we fixed this problem in the repo but had not pushed it out to PyPi).

**Comment by 59e5aaf4** [comment]
Works fine with the latest version indeed, RAM grows up to a standard 800MB (order of magnitude of non-profiled operations), and it doesn't explode. I however noticed that "--profile-interval 15" doesn't seem to work any more, before it would write the .html every 15s. Not related to the current issue though.

Thanks !

**Comment by Akashdesarda** [comment]
@59e5aaf4 You are right, after the update the SIGBART issue seems to be fixed but like you, I was also using --profile-interval 1 which doesn't seem to be working. But thanks to @emeryberger & team scalene for the fix. You are awesome.

**Comment by emeryberger** [comment]
Coincidentally, I also found the `--profile-interval` bug and fixed it in the repository. Will push a new version to PyPi in a few minutes.

**Comment by emeryberger** [comment]
Fix posted, please update (1.3.6).
